src/modules/agora/AgoraClient.ts

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+
+    The class that recover data
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+import axios, { AxiosInstance, AxiosResponse, AxiosError } from 'axios';
+import URI from "urijs";
+
+/**
+ * The class that recover data
+ */
+export class AgoraClient
+{
+    /**
+     * The instance of the axios
+     */
+    private client: AxiosInstance;
+
+    /**
+     * The URL encoder
+     */
+    private uri: URI;
+
+    /**
+     * Constructor
+     * @param host - The network address to connect to the Agora
+     * @param port - The network port to connect to the Agora
+     */
+    constructor (host: string, port: string)
+    {
+        this.client = axios.create();
+        this.client.defaults.timeout = 10000;
+        this.uri = URI(host).port(port);
+    }
+
+    /**
+     * Requests and receives data needed for recovery from the Agora node.
+     * @param block_height - The height of the block
+     * @param max_blocks - The maximum number of block to request
+     */
+    public requestBlocks (block_height: number, max_blocks: number): Promise<Array<any>>
+    {
+        return new Promise<Array<any>>((resolve, reject) =>
+        {
+            this.uri
+                .directory("blocks_from")
+                .setSearch("block_height", block_height.toString())
+                .setSearch("max_blocks", max_blocks.toString());
```

Comment:
If I understand correctly, this modifies the `uri` member ? Doesn't that mean that this function is not re-entrant, which is at odds with its `Promise`-based approach ?

Reply:
The address and port of the client need not be preset. Because I have to enter the entire URL every time I call this.client.get.
So as an alternative, I set the address and port on the URL.

I checked in advance through the test.
I have confirmed that it is reset.
```
describe ('Test of Stoa API Server', () => {
    let host: string = 'http://localhost';
    let port: string = '3837';

    it('Test of URI', () => {
        let uri = URI(host)
            .port(port);

        uri
            .directory("1")
            .setSearch("block_height", "1")
            .setSearch("max_blocks", "1");
        assert.strictEqual(uri.toString(), "http://localhost:3837/1/?block_height=1&max_blocks=1");

        uri
            .directory("2")
            .setSearch("block_height", "2")
            .setSearch("max_blocks", "2");
        assert.strictEqual(uri.toString(), "http://localhost:3837/2/?block_height=2&max_blocks=2");
    });
});
```

Reply:
@Geod24 I'll check and ask for a review because there's something I need to revise.

Reply:
@Geod24 That's right. The value of A can be changed by other repeated calls.
So I changed it back to a local variable
Ready of review